devito/visitors.py

```diff
@@ -143,6 +149,51 @@ def always_rebuild(self, o, *args, **kwargs):
         return o._rebuild(*new_ops, **okwargs)
 
 
+class PrintAST(Visitor):
+
+    _depth = 0
```

Comment:
Actually, you might just get rid of it and simply propagate the depth as a parameter along the call tree